tests/test_trainer.py

```diff
@@ -97,16 +97,16 @@ def test_snapshoting():
         optimizer,
         train_data=data,
         validation_data=None,
-        callbacks=[core.CheckpointSaver(checkpoint_path=CHECKPOINT_PATH)],
+        callbacks=[core.CheckpointSaver(checkpoint_path=CHECKPOINT_PATH, max_checkpoints=2)],
     )
-    trainer.train(2)
-    assert (CHECKPOINT_PATH / Path("1.tar")).exists()
-    assert (CHECKPOINT_PATH / Path("2.tar")).exists()
+    trainer.train(n_epochs=6)
+    assert (CHECKPOINT_PATH / Path("5.tar")).exists()
+    assert (CHECKPOINT_PATH / Path("6.tar")).exists()
     assert (CHECKPOINT_PATH / Path("final.tar")).exists()
     del trainer
     trainer = core.Trainer(game, optimizer, train_data=data)  # Re-instantiate trainer
     trainer.load_from_latest(CHECKPOINT_PATH)
-    assert trainer.start_epoch == 2
+    assert trainer.start_epoch == 6
```

Comment:
how about an additional check that *only* checkpoint_5 and 6 are present in the folder but in a new test?

Reply:
How do you do that?

Reply:
You could leave the task as it was (without max_checkpoints and with two epochs/checkpoints only) and have another test that specifically check the max_checkpoints methods (you can use the current new test with max_checkpoints)

Reply:
done in 
dc6fd3a

Reply:
The latest test is just a copy-paste of the the 2-epoch one but checking that even when using max_checkpoints training with 6-epochs the model stores checkpoint_{5, 6, final}. An ideal test would be checking that *only* those checkpoints are present. Thus making sure the the others are correctly deleted